homework03/test/examples/Main.hs

```diff
@@ -1,15 +1,39 @@
-module Main
-  ( main
-  )
-  where
-
--- doctest
-import qualified Test.DocTest as DocTest
+module Main 
+  ( main )
+where
 
+import Golf
+import Test.Hspec
 
 main :: IO ()
-main =
-  DocTest.doctest
-    [ "-isrc"
-    , "src/Golf.hs"
-    ]
+main = hspec $ do
+
+  describe "skips" $ do
+    it "skips list of Chars (length 4)" $
+      skips "ABCD" `shouldBe` ["ABCD", "BD", "C", "D"]
+      
+    it "skips list of Chars (length 6)" $
+      skips "hello!" `shouldBe` ["hello!", "el!", "l!", "l", "o", "!"]
+      
+    it "skips list of one element" $ 
+      skips [1] `shouldBe` [[1]]
+
+    it "skips list of Booleans" $
+      skips [True,False] `shouldBe` [[True,False], [False]]
+
+    it "skips empty list" $
+      skips ([] :: [String]) `shouldBe` ([] :: [[String]])
+      
+  describe "localMaxima" $ do
+    it "finds local maxima and gets 2 elements" $
+      localMaxima [2,9,5,6,1] `shouldBe` [9,6]
+
+    it "finds local maxima and gets 1 elements" $
+      localMaxima [2,3,4,1,5] `shouldBe` [4]
+
+    it "finds local maxima and not gets elements" $
```

Comment:
@jmorocho "find local maxima and doesn't get elements"?

In addition, I think we're missing some context functions here. Why in some case do you get one element or two? 

```haskell 
context "list.... "
  it "gets 2 elements"
```